src/nemos/basis/_basis.py

```diff
@@ -480,9 +484,35 @@ def __mul__(self, other: Basis) -> MultiplicativeBasis:
         :
             The resulting Basis object.
         """
+        if isinstance(other, int):
+            if other <= 0:
+                raise ValueError(
+                    "Basis multiplication error. Integer multiplicative factor must be positive, "
+                    f"{other} provided instead."
+                )
+            elif not all(
+                b._has_default_label for _, b in generate_basis_label_pair(self)
+            ):
+                raise ValueError(
+                    "Cannot multiply by an integer a basis including a user-defined labels."
```

Comment:
```suggestion
                    "Cannot multiply by an integer a basis including a user-defined labels (because then they won't be unique). Set labels after multiplication."
```